sui-execution/next-vm/sui-adapter/src/execution_engine.rs

```diff
@@ -521,14 +521,12 @@ mod checked {
 
                 for genesis_object in objects {
                     match genesis_object {
-                        sui_types::transaction::GenesisObject::RawObject { data, owner } => {
-                            let object = ObjectInner {
+                        GenesisObject::RawObject { data, owner } => {
+                            temporary_store.create_object(Object::new_from_genesis(
```

Comment:
Ideally, some tests for this new function `new_from_genesis` might be useful?

Reply:
I would really struggle to find some behaviour to test in this code path that isn't captured by the existing authority tests around genesis -- it is literally a constructor call that defaults one of the arguments to zero.